WooCommerce/Classes/Authentication/RoleEligibilityUseCase.swift

```diff
@@ -35,8 +35,11 @@ final class RoleEligibilityUseCase {
 
     /// The last eligibility error info encountered by the user.
     private var lastErrorInfo: EligibilityErrorInfo? {
-        didSet {
-            defaults.setValue(lastErrorInfo?.toDictionary(), forKey: Constants.eligibilityErrorInfoKey)
+        get {
+            loadLastErrorInfo()
+        }
+        set {
+            defaults.setValue(newValue?.toDictionary(), forKey: Constants.eligibilityErrorInfoKey)
```

Comment:
⚠️  We try to not save more things to user defaults. For settings like these, we have `GeneralAppSettings` which is accessible through a `Yosemite` store.

Reply:
Noted. Will look into this and move the implementation to `GeneralAppSettings` instead. 👍

Reply:
Addressed in #4665 !